sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetTest.scala

```diff
@@ -62,13 +62,21 @@ private[sql] trait ParquetTest extends FileBasedDataSourceTest {
       (data: Seq[T])
       (f: String => Unit): Unit = withDataSourceFile(data)(f)
 
+  protected def toDF[T <: Product: ClassTag: TypeTag](data: Seq[T]): DataFrame = {
```

Comment:
instead of adding this new util function, shall we add an overload of `withParquetDataFrame` that still takes `Seq`? Then we can avoid changing some test cases.

Reply:
I was thinking to do so, but surprisingly, overloading 
```scala
protected def withParquetDataFrame[T <: Product: ClassTag: TypeTag]
      (data: Seq[T], testVectorized: Boolean = true)
      (f: DataFrame => Unit): Unit
```
and 
```scala
protected def withParquetDataFrame(df: DataFrame, testVectorized: Boolean = true)
   (f: DataFrame => Unit): Unit
```
is not allowed in Scala.

Reply:
The error is
```scala
Error:(45, 20) in trait ParquetTest, multiple overloaded alternatives of method withParquetDataFrame define default arguments.
private[sql] trait ParquetTest extends FileBasedDataSourceTest {
```